docs/tutorials/build-your-first-blazor-app.md

```diff
@@ -0,0 +1,485 @@
+
+```cshtml
+@page "/fetchdata"
+@inject HttpClient Http
+```
+
+The `FetchData` component uses the injected `HttpClient` to retrieve JSON data from the server when the component is initialized. Under the covers, the `HttpClient` provided by the Blazor runtime is implemented using JavaScript interop to call the underlying browser's Fetch API to send the request (from C#, it's possible to call any JavaScript library or browser API). The retrieved data is deserialized into the `forecasts` C# variable as an array of `WeatherForecast` objects.
+
+```cshtml
+@functions {
+    WeatherForecast[] forecasts;
+
+    protected override async Task OnInitAsync()
+    {
+        forecasts = await Http.GetJsonAsync<WeatherForecast[]>("/sample-data/weather.json");
+    }
+
+    class WeatherForecast
+    {
+        public DateTime Date { get; set; }
+        public int TemperatureC { get; set; }
+        public int TemperatureF { get; set; }
+        public string Summary { get; set; }
+    }
+}
+```
+
+A `@foreach` loop is used to render each forecast instance as a row in the weather table.
+
+```cshtml
+<table class="table">
+    <thead>
+        <tr>
+            <th>Date</th>
+            <th>Temp. (C)</th>
+            <th>Temp. (F)</th>
+            <th>Summary</th>
+        </tr>
+    </thead>
+    <tbody>
+        @foreach (var forecast in forecasts)
+        {
+            <tr>
+                <td>@forecast.Date.ToShortDateString()</td>
+                <td>@forecast.TemperatureC</td>
+                <td>@forecast.TemperatureF</td>
+                <td>@forecast.Summary</td>
+            </tr>
+        }
+    </tbody>
+</table>
+```
+
+## Build a todo list
+
+Add a new page to the app that implements a simple todo list.
+
+1. Add a Todo page to the app by right-clicking on the *Pages* folder and selecting **Add** > **New item** > **Razor View**. Name the file *Todo.cshtml* and select **OK**.
+
+    ![Add Razor View](https://user-images.githubusercontent.com/1874516/38331329-cff9e54a-3807-11e8-9cfe-2885678bb58a.png)
+
+1. Replace the content of the *Todo.cshtml* file with some initial markup.
+
+    ```cshtml
+    @page "/todo"
+
+    <h1>Todo</h1>
+    ```
+
+1. Add the Todo page to the navigation bar by updating *Shared/NavMenu.cshtml*. Add a `NavLink` for the Todo page by adding the following list item markup below the existing list items.
+
+    ```cshtml
+    <li>
+        <NavLink href="/todo">
+            <span class='glyphicon glyphicon-th-list'></span> Todo
+        </NavLink>
+    </li>
+    ```
+
+1. Build and run the app. See the new Todo page.
+
+    ![Blazor todo start](https://msdnshared.blob.core.windows.net/media/2018/03/blazor-todo-start.png)
+
+1. Add a class to represent the todo items by right-clicking on the project and selecting **Add** > **Class**. Name the file *TodoItem.cs* and select **OK**.
```

Comment:
Again, let's not use the item template. Just tell them to add a *TodoItem.cs* file to the root of the project. The benefit of this is that the instruction are more generic and apply to xplat usage.